### PR TITLE
Public fixes

### PR DESCRIPTION
**Pull request by Meruja**
Fix API store open redirection vulnerability.(https://github.com/wso2-support/carbon-apimgt/pull/2395) and User email encoding in password reset process (https://github.com/wso2-support/carbon-apimgt/pull/2437)

### CODE REVIEW

File: features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/store/jagg/jagg.jag [hunk 1]
```diff
@@ -935,6 +935,15 @@ var jagg = jagg || (function () {
         return content.replace(startPattern, "&lt;script &gt;").replace(endPattern, "&lt;\script&gt;");
     };
 
+    var startsWith = function(path, comparator){
+        for(var i=0;i<comparator.length;i++) {
```
Comment (isharac) [inline features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/store/jagg/jagg.jag:939]: need spaces

File: features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/store/jagg/jagg.jag [hunk 2]
```diff
@@ -935,6 +935,15 @@ var jagg = jagg || (function () {
...
+            if(path[i] != comparator[i]){
```
Comment (isharac) [inline features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/store/jagg/jagg.jag:940]: need space

File: features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/store/jagg/jaggery_acs.jag [hunk 3]
```diff
@@ -265,8 +265,14 @@
                         if (MultitenantConstants.SUPER_TENANT_DOMAIN_NAME !== domainName) {
                             jagg.module("manager").loadTenantAxisConfiguration(String(domainName));
                         }
-                        response.sendRedirect(requestedURI);
-                        return;
+                        var siteContext = jagg.getSiteContext();
+                        if (jagg.startsWith(requestedURI, siteContext)){
```
Comment (isharac) [inline features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/store/jagg/jaggery_acs.jag:269]: need space

**Review by isharac — CHANGES_REQUESTED** [review]

File: features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/store/jagg/jaggery_acs.jag (hunk 3 above)
Comment (Meruja) [reply features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/store/jagg/jaggery_acs.jag]: Fixed with 74c1913

File: features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/store/jagg/jagg.jag (hunk 2 above)
Comment (Meruja) [reply features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/store/jagg/jagg.jag]: Fixed with 74c1913

File: features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/store/jagg/jagg.jag (hunk 1 above)
Comment (Meruja) [reply features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/store/jagg/jagg.jag]: Fixed with 74c1913

**Review by isharac — APPROVED** [review]

**Pull request merged**